src/meta/proto-conv/tests/it/user_stage.rs

```diff
@@ -53,6 +53,52 @@ fn test_user_stage_oss_latest() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_user_stage_fs_v21() -> anyhow::Result<()> {
+    // Encoded data of version 21 of user_stage_fs:
+    // It is generated with common::test_pb_from_to.
+    let user_stage_fs_v21 = vec![
+        10, 17, 102, 115, 58, 47, 47, 100, 105, 114, 47, 116, 111, 47, 102, 105, 108, 101, 115, 26,
+        25, 10, 23, 18, 21, 10, 13, 47, 100, 105, 114, 47, 116, 111, 47, 102, 105, 108, 101, 115,
+        160, 6, 21, 168, 6, 1, 34, 33, 8, 1, 16, 128, 8, 26, 1, 124, 34, 2, 47, 47, 40, 2, 50, 1,
+        92, 58, 3, 114, 111, 119, 66, 3, 78, 97, 78, 160, 6, 21, 168, 6, 1, 42, 10, 10, 3, 32, 154,
+        5, 16, 142, 8, 24, 1, 50, 4, 116, 101, 115, 116, 160, 6, 21, 168, 6, 1,
+    ];
+
+    let want = mt::UserStageInfo {
+        stage_name: "fs://dir/to/files".to_string(),
+        stage_type: mt::StageType::LegacyInternal,
+        stage_params: mt::StageParams {
+            storage: StorageParams::Fs(StorageFsConfig {
+                root: "/dir/to/files".to_string(),
+            }),
+        },
+        file_format_options: mt::FileFormatOptions {
+            format: mt::StageFileFormatType::Json,
+            skip_header: 1024,
+            field_delimiter: "|".to_string(),
+            record_delimiter: "//".to_string(),
+            nan_display: "NaN".to_string(),
+            compression: mt::StageFileCompression::Bz2,
+            escape: "\\".to_string(),
+            row_tag: "row".to_string(),
+        },
+        copy_options: mt::CopyOptions {
```

Comment:
If UserStageInfo is not affected by this change, you can just test FileFormatOptions

Reply:
As the `user.proto` version already increased, I think testing full UserStageInfo is OK here.